### PR TITLE
Fix/prepare statements

### PR DESCRIPTION
**Pull request by MRuy**

### CODE REVIEW

File: src/routes/getTopUsers.ts [hunk 1]
```diff
@@ -25,14 +25,14 @@ async function generateTopUsersStats(sortBy: string, categoryStatsEnabled: boole
...
-        SUM(((CASE WHEN "sponsorTimes"."endTime" - "sponsorTimes"."startTime" > ${maxRewardTimePerSegmentInSeconds} THEN ${maxRewardTimePerSegmentInSeconds} ELSE "sponsorTimes"."endTime" - "sponsorTimes"."startTime" END) / 60) * "sponsorTimes"."views") as "minutesSaved",
+        SUM(((CASE WHEN "sponsorTimes"."endTime" - "sponsorTimes"."startTime" > ? THEN ? ELSE "sponsorTimes"."endTime" - "sponsorTimes"."startTime" END) / 60) * "sponsorTimes"."views") as "minutesSaved",
         SUM("votes") as "userVotes", ` +
         additionalFields +
         `IFNULL("userNames"."userName", "sponsorTimes"."userID") as "userName" FROM "sponsorTimes" LEFT JOIN "userNames" ON "sponsorTimes"."userID"="userNames"."userID"
         LEFT JOIN "privateDB"."shadowBannedUsers" ON "sponsorTimes"."userID"="privateDB"."shadowBannedUsers"."userID"
         WHERE "sponsorTimes"."votes" > -1 AND "sponsorTimes"."shadowHidden" != 1 AND "privateDB"."shadowBannedUsers"."userID" IS NULL
         GROUP BY IFNULL("userName", "sponsorTimes"."userID") HAVING "userVotes" > 20
-        ORDER BY "${sortBy}" DESC LIMIT 100`, []);
```
Comment (ajayyy) [inline src/routes/getTopUsers.ts:35]: Last time I tried this, it did not work. Have you tested this?

Comment (MRuy) [reply src/routes/getTopUsers.ts:35]: No it doesnt work, it will be fixed in #233

**Comment by ajayyy** [comment]
Not sure the state of this, was it reliant on #233?

File: src/routes/getTopUsers.ts [hunk 2]
```diff
@@ -26,14 +26,14 @@ async function generateTopUsersStats(sortBy: string, categoryStatsEnabled: boole
...
         SUM("votes") as "userVotes", ` +
         additionalFields +
         `COALESCE("userNames"."userName", "sponsorTimes"."userID") as "userName" FROM "sponsorTimes" LEFT JOIN "userNames" ON "sponsorTimes"."userID"="userNames"."userID"
-        LEFT JOIN "shadowBannedUsers" ON "sponsorTimes"."userID"="shadowBannedUsers"."userID"
+        LEFT JOIN "privateDB"."shadowBannedUsers" ON "sponsorTimes"."userID"="privateDB"."shadowBannedUsers"."userID"
         WHERE "sponsorTimes"."votes" > -1 AND "sponsorTimes"."shadowHidden" != 1 AND "shadowBannedUsers"."userID" IS NULL
-        GROUP BY COALESCE("userName", "sponsorTimes"."userID") HAVING SUM("votes") > 20
-        ORDER BY "${sortBy}" DESC LIMIT 100`, []);
+        GROUP BY COALESCE("userName", "sponsorTimes"."userID") HAVING "userVotes" > 20
+        ORDER BY ? DESC LIMIT 100`, [maxRewardTimePerSegmentInSeconds, maxRewardTimePerSegmentInSeconds, sortBy]);
```
Comment (ajayyy) [inline src/routes/getTopUsers.ts:36]: Need to test this on postgres

File: src/routes/getTopUsers.ts [hunk 3]
```diff
@@ -26,14 +26,14 @@ async function generateTopUsersStats(sortBy: string, categoryStatsEnabled: boole
...
+        SUM(((CASE WHEN "sponsorTimes"."endTime" - "sponsorTimes"."startTime" > ? THEN ? ELSE "sponsorTimes"."endTime" - "sponsorTimes"."startTime" END) / 60) * "sponsorTimes"."views") as "minutesSaved",
         SUM("votes") as "userVotes", ` +
         additionalFields +
         `COALESCE("userNames"."userName", "sponsorTimes"."userID") as "userName" FROM "sponsorTimes" LEFT JOIN "userNames" ON "sponsorTimes"."userID"="userNames"."userID"
-        LEFT JOIN "shadowBannedUsers" ON "sponsorTimes"."userID"="shadowBannedUsers"."userID"
+        LEFT JOIN "privateDB"."shadowBannedUsers" ON "sponsorTimes"."userID"="privateDB"."shadowBannedUsers"."userID"
         WHERE "sponsorTimes"."votes" > -1 AND "sponsorTimes"."shadowHidden" != 1 AND "shadowBannedUsers"."userID" IS NULL
-        GROUP BY COALESCE("userName", "sponsorTimes"."userID") HAVING SUM("votes") > 20
-        ORDER BY "${sortBy}" DESC LIMIT 100`, []);
+        GROUP BY COALESCE("userName", "sponsorTimes"."userID") HAVING "userVotes" > 20
```
Comment (ajayyy) [inline src/routes/getTopUsers.ts:35]: Could you revert this change, postgres doesn't let you reuse definitions

File: src/routes/getTopUsers.ts [hunk 4]
```diff
@@ -26,14 +26,14 @@ async function generateTopUsersStats(sortBy: string, categoryStatsEnabled: boole
     }
 
     const rows = await db.prepare('all', `SELECT COUNT(*) as "totalSubmissions", SUM(views) as "viewCount",
-        SUM(((CASE WHEN "sponsorTimes"."endTime" - "sponsorTimes"."startTime" > ${maxRewardTimePerSegmentInSeconds} THEN ${maxRewardTimePerSegmentInSeconds} ELSE "sponsorTimes"."endTime" - "sponsorTimes"."startTime" END) / 60) * "sponsorTimes"."views") as "minutesSaved",
+        SUM(((CASE WHEN "sponsorTimes"."endTime" - "sponsorTimes"."startTime" > ? THEN ? ELSE "sponsorTimes"."endTime" - "sponsorTimes"."startTime" END) / 60) * "sponsorTimes"."views") as "minutesSaved",
         SUM("votes") as "userVotes", ` +
         additionalFields +
         `COALESCE("userNames"."userName", "sponsorTimes"."userID") as "userName" FROM "sponsorTimes" LEFT JOIN "userNames" ON "sponsorTimes"."userID"="userNames"."userID"
-        LEFT JOIN "shadowBannedUsers" ON "sponsorTimes"."userID"="shadowBannedUsers"."userID"
+        LEFT JOIN "privateDB"."shadowBannedUsers" ON "sponsorTimes"."userID"="privateDB"."shadowBannedUsers"."userID"
```
Comment (ajayyy) [inline src/routes/getTopUsers.ts:33]: Could you revert this change?

**Review by ajayyy — CHANGES_REQUESTED** [review]

**Review by ajayyy — APPROVED** [review]
